dotCMS/src/main/java/com/dotmarketing/portlets/templates/business/TemplateAPIImpl.java

```diff
@@ -258,24 +262,47 @@ public Template saveTemplate(Template template, Host destination, User user, boo
 	public List<Container> getContainersInTemplate(final Template template, final User user, final boolean respectFrontendRoles)
             throws DotDataException, DotSecurityException {
 
-		final TemplateLayout layout = DotTemplateTool.themeLayout(template.getInode());
-		final List<String> containersId = layout.getContainersId();
+        final List<Container> containers = new ArrayList<>();
 
-		final List<Container> containers = new ArrayList<>();
+	    if (template.isDrawed()){
+            final TemplateLayout layout     = DotTemplateTool.themeLayout(template.getInode());
+            final List<String> containersId = layout.getContainersId();
 
-		for (final String cont : containersId) {
-			final Container container = APILocator.getContainerAPI().getWorkingContainerById(cont, user, false);
+            for (final String cont : containersId) {
+                final Container container = APILocator.getContainerAPI().getWorkingContainerById(cont, user, false);
 
-			if(container==null) {
-				continue;
-			}
+                if (container == null) {
+                    continue;
+                }
 
-			containers.add(container);
-		}
+                containers.add(container);
+            }
+        }
+
+        // this is a light weight search for pages that use this template
+        List<ContentletSearch> pages = APILocator.getContentletAPIImpl()
+            .searchIndex("+_all:" + template.getIdentifier()
+                + " +baseType:" + BaseContentType.HTMLPAGE.getType(),
+                100, 0, null, user, respectFrontendRoles);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Assign this magic number 100 to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS109)